### PR TITLE
Move `useAuth` to react-core

### PR DESCRIPTION
**Pull request by wlee221**
#### Description of changes

Moves `useAuth` to react core, as React Native will this util.

This is re-exported at `@aws-amplify/ui-react/internal` to not break existing usage.

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by calebpollman — APPROVED** [review]
🚢

File: packages/react/src/internal.tsx [hunk 1]
```diff
@@ -1,4 +1,5 @@
-export * from './hooks/useAuth';
+export { useAuth } from '@aws-amplify/ui-react-core';
```
Comment (wlee221) [inline packages/react/src/internal.tsx:1]: This is where I re-export `useAuth` from `@aws-amplify/ui-react`.

**Review by reesscot — APPROVED** [review]
LGTM

File: packages/react/src/internal.tsx (hunk 1 above)
Comment (calebpollman) [reply packages/react/src/internal.tsx:1]:
```suggestion
// re-export `useAuth` for Studio usage
export { useAuth } from '@aws-amplify/ui-react-core';
```

**Review by calebpollman — APPROVED** [review]

**Pull request merged**